katdal/applycal.py

```diff
@@ -342,6 +379,7 @@ def calc_correction_per_input(cache, name, inp, product_type):
             correction_sensor = calc_bandpass_correction(product_sensor, index,
                                                          data_freqs, cal_freqs)
         elif product_type == 'G':
+            product_sensor = calibrate_flux(product_sensor, targets, gaincal_fluxes)
```

Comment:
Is this the optimal place to scale the gains?

Reply:
Ideally one could move it earlier, to when the G product is extracted from telstate, like a custom virtual sensor for `'Calibration/Products/l1/G'` instead of the usual `indirect_cal_product` function, but I consider it one hoop too many unless the performance benefit is clear.